promoted_python_delivery_client/client/serde.py

```diff
@@ -0,0 +1,61 @@
+import ujson
+from promoted_python_delivery_client.model.request import Request
+from promoted_python_delivery_client.model.response import Response
+from promoted_python_delivery_client.model.log_request import LogRequest
+
+
+delivery_request_schema = Request.schema()  # type: ignore this is from dataclass_json
+log_request_schema = LogRequest.schema()  # type: ignore this is from dataclass_json
+
+
+# DEPRECATED use delivery_request_to_json_3
+def delivery_request_to_json(req: Request) -> str:
+    return req.to_json()  # type: ignore this is from dataclass_json
+
+
+# DEPRECATED use delivery_request_to_json_3
+def delivery_request_to_json_2(req: Request) -> str:
+    return ujson.dumps(req.to_dict(encode_json=False))  # type: ignore this is from dataclass_json
+
+
+# This one profiles the fastest, the others are here for reference purposes.
+def delivery_request_to_json_3(req: Request) -> str:
+    kvs = delivery_request_schema.dump(req)  # type: ignore this is from dataclass_json
+    return ujson.dumps(_clean_empty(kvs))
+
+
+# DEPRECATED use log_request_to_json_3
+def log_request_to_json(req: LogRequest) -> str:
+    return req.to_json()  # type: ignore this is from dataclass_json
+
+
+# DEPRECATED use log_request_to_json_3
+def log_request_to_json_2(req: LogRequest) -> str:
+    return ujson.dumps(req.to_dict(encode_json=False))  # type: ignore this is from dataclass_json
+
+
+# This one profiles the fastest, the others are here for reference purposes.
+def log_request_to_json_3(req: LogRequest) -> str:
+    kvs = log_request_schema.dump(req)  # type: ignore this is from dataclass_json
+    return ujson.dumps(_clean_empty(kvs))
+
+
+def delivery_response_from_json(payload: bytes) -> Response:
+    return Response.from_json(payload)  # type: ignore this is from dataclass_json
+
+
+def delivery_reponse_from_json_2(payload: bytes) -> Response:
+    kvs = ujson.loads(payload)
+    return Response.from_dict(kvs)  # type: ignore this is from dataclass_json
+
+
+def _clean_empty(d):
+    if isinstance(d, dict):
+        return {
+            k: v
+            for k, v in ((k, _clean_empty(v)) for k, v in d.items())
+            if v
```

Comment:
Hmm, should this explicitly just check None?

Also, might have a weird impact if this is done on the properties fields.  I figured we'd want to distinguish between a request property being set to false vs not set at all.

Reply:
I changed it to this, please have a look:
```
if (v is not None) or inside_props_struct
```

I'm not sure passing nulls/Nones as properties struct values is something we should be encouraging in most cases (because it would be easy to create bugs that explode payload size), but now the code supports it (I added a unit test).